### DISCUSSION TITLE
Is there a way to create widgets for the top bar?

### DISCUSSION THREAD

**Discussion by fkrahe** [pop-os/cosmic-epoch#1094] [Ideas]
Hello, I'm a developer, but no a Linux developer. I would like to have some widgets in the top bar, like a System Monitor or something like that. Is there support to make these? An API or some guides?

**Comment by jacobgkau** [comment]
Yes, the top panel's contents are made up of applets. Applets run in their own process and are typically written in Rust. The default ones are here: https://github.com/pop-os/cosmic-applets 

You're welcome to join the ~development chat room on https://chat.pop-os.org to ask questions about applet development. Some others have already made third-party applets. An unofficial applet template from one of those third-party developers can be found here: https://github.com/edfloreshz/cosmic-applet-template
